### PR TITLE
Move quant ops to utils.py

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
We had a lot of "quant primitive" ops that can be expressed with more primitive ops, so these ops are more of a helper functions now, so we moved them to torchao.quantization.utils

we should be able to further deprecate some of the ops after we deprecate subclasses and refactor smoothquant etc. in the future

Also moved TORCH_VERSION_AFTER_{2_2/2_3/2_4} from torchao.quantization.utils to torchao.utils

After the move we have the following ops in quant_primitives.py:

```
    "safe_int_mm",
    "int_scaled_matmul",
    "choose_qparams_affine",
    "quantize_affine",
    "dequantize_affine",
```
Test Plan:
python test/integration/test_integration.py
python test/quantization/test_quant_api.py
python test/quantization/test_quant_primitives.py
python test/quantization/test_qat.py

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/quantization/test_qat.py [hunk 1]
```diff
@@ -18,8 +18,8 @@
     fake_quantize_per_channel_group,
     fake_quantize_per_token,
 )
-from torchao.quantization.quant_primitives import get_group_qparams_symmetric
-from torchao.quantization.utils import TORCH_VERSION_AFTER_2_4
+from torchao.quantization.utils import get_group_qparams_symmetric
```
Comment (msaroufim) [inline test/quantization/test_qat.py:21]: I'm having trouble figuring out what should be in quant_primitives vs utils

Comment (jerryzh168) [reply test/quantization/test_qat.py:21]: functions in utils are mostly helper functions that calls the quant_primitive ops with some fixed parameters, e.g. 

quantize_affine can support: symmetric/asymmetric, per tensor/group/channel etc., int8/int4/int3

helper function in utils can be: int8_symmetric_per_tensor_quant that calls quantize_affine op with fixed settings

File: torchao/utils.py [hunk 2]
```diff
@@ -65,3 +66,18 @@ def find_multiple(n: int, *args: Tuple[int]) -> int:
     if n % k == 0:
         return n
     return n + k - (n % k)
+
+if version.parse(torch.__version__) >= version.parse("2.4.0.dev"):
```
Comment (msaroufim) [inline torchao/utils.py:70]: Thanks for this! I wish for this PR we had seperated out the easy utils changes from the more controversial BC changes. Would make this easier to merge

File: test/integration/test_integration.py [hunk 3]
```diff
@@ -90,6 +93,120 @@
 
 COMMON_DEVICE_DTYPE = list(itertools.product(COMMON_DEVICES, COMMON_DTYPES)).copy()
 
+# deprecated quant primitive ops
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply test/integration/test_integration.py:96]: these are just kept for testing (for quant primitive ops), yeah I can delete as well

File: torchao/utils.py (hunk 2 above)
Comment (jerryzh168) [reply torchao/utils.py:70]: OK sure

Comment (jerryzh168) [reply torchao/utils.py:70]: https://github.com/pytorch/ao/pull/337

**Review by msaroufim — APPROVED** [review]

**Pull request merged**